### ISSUE TITLE
What is the standard way to bring arrays into standard order?

### ISSUE THREAD

**Issue by ExpHP** [bluss/ndarray#441]
Suppose I have an arbitrary array; any model of ownership, any number of dimensions, possibly non-contiguous, with strides in whatever order.  How do I get an owned array in standard layout?  And could it be better documented?

Based on the documentation of ArrayBase alone, it seems to me that the following is the only safe bet: 

```rust
Array::from_shape_vec(arr.raw_dim(), arr.iter().cloned().collect())
```

**Comment by jturner314** [comment]
Yeah, that's the simplest way, I think. The other obvious things to try (`.to_owned()` and `.map()`/`.mapv()`) don't alter the layout of contiguous arrays, so they don't help in this case.

Something like this might be faster for handling contiguous arrays in standard layout, but you'd have to benchmark to know for sure:

```rust
if let Some(slc) = self.as_slice() {
    Array::from_shape_vec(arr.raw_dim(), slc.to_vec())
} else {
    Array::from_shape_vec(arr.raw_dim(), arr.iter().cloned().collect())
}
```

If your element type is `Copy`, an alternative approach is to allocate a new array with `::uninitialized()` and then `.assign()` the data.

> And could it be better documented?

I've created #442.

**Comment by jturner314** [comment]
Resolved by #442.

It would be nice in the future to provide a more concise way to convert between memory layouts, possibly using something like the `Cow`-like types described in #390.
